db/compaction/compaction_picker_fifo.cc

```diff
@@ -370,8 +370,11 @@ Compaction* FIFOCompactionPicker::PickTemperatureChangeCompaction(
         return nullptr;
       }
       uint64_t est_newest_key_time = cur_file->TryGetNewestKeyTime(prev_file);
-      if (est_newest_key_time == kUnknownNewestKeyTime ||
-          est_newest_key_time > create_time_threshold) {
+      // Newer file could have newest_key_time populated
```

Comment:
@cbi42 After thinking through this, I do not think this change is actually necessary. Although it is reasonable to expect `newest_key_time` to only be found in the younger files,`TryGetNewestKeyTime` uses `prev_file->TryGetOldestAncesterTime()` as a fallback if the `newest_key_time` is not available.

If we hit `est_newest_key_time == kUnknownNewestKeyTime`, that means that neither the `newest_key_time` nor the `oldest_ancester_time` (`creation_time`) of the previous file were available.

`TryGetOldestAncesterTime()` was the old logic being used, so I don't think we would expect those to be missing.

Thus, I don't think we would end up hitting the case where a file's `newest_key_time` and `oldest_ancester_time` (`creation_time`) are all unknown, but the next (younger) file has one of these values.

Thoughts?

Reply:
> TryGetOldestAncesterTime() was the old logic being used, so I don't think we would expect those to be missing.

I think it may still happen in rare cases, e.g. older versions of Rocksdb is used or clock_->GetCurrentTime() fails.

Reply:
> I think it may still happen in rare cases

Gotcha, in that case, I opened this PR for review